[4.0] Skipto

brianteeman opened the pull request:
Initial commit - still a work in progress

@dgrammatiko Please advise/help on the addScriptOptions

rigin committed:
Add a note field from the #__fields_groups table to the field list query

Commit:
[plg_user_profile] Add RTL check to ToS field

Hackwar committed:
Fix OpenSearch implementation

Commit:
[mod_articles_category] Showon

Commit:
Add showon to levels

Commit:
Update string

Commit:
Remove CSS

Commit:
Merge pull request #20950 from SharkyKZ/modArticlesCategoryShowon

[mod_articles_category] Showon

Commit:
Merge pull request #20937 from Hackwar/patch-28

Fix OpenSearch implementation

Commit:
Merge pull request #20894 from SharkyKZ/plg_user_profile_rtl

[plg_user_profile] Add RTL check to ToS field

Commit:
Add security issue template

Commit:
Formatting

Commit:
Let the table check the data before storing.

Commit:
[Templates] Escape sitename (#21008)

Commit:
Checking return value and catching whole stuff

Commit:
Updating Bosnian installation language files

infograf768 committed:
Merge pull request #21080 from Bakual/FixSampledataMenuCreation

Fix sampledata menu creation

jeckodevelopment committed:
Updated 3.9-dev and added 3.10-dev (#21097)

brianteeman committed:
Exceptions in Joomla\CMS\Table\Usergroup refer to categories (#21098)

* Exceptions in Joomla\CMS\Table\Usergroup refer to categories

Pull Request for Issue #21092 .

### Summary of Changes
remove todo that is clearly a copy paste error from somewhere else
change exception message to usergroups from catergory

NOTE the first exception` if ($this->id == 0)`
I am not sure if that should even be present - doesn't look like it to me - please advise

* Update Usergroup.php

Commit:
Deprecate sef_advanced strings (#21116)

Commit:
[com_content] Featured articles tag filtering (#21138)

Commit:
Update ModuleHelper.php (#20273)

Commit:
The JPATH_ constants are not magic strings (#21151)

Commit:
Remove if/else in user debug models that can never hit else conditions (#21188)

Commit:
[com_content] Featured articles access filter (#21168)

wojsmol committed:
Media manager doesn't obey relative pathes for video files (#21156)

* correct video url

fixes https://github.com/joomla/joomla-cms/issues/21145#issuecomment-405484769

* fix

csthomas committed:
Speed up regex in emailcloak plugin (#20956)

* Speed up regex in emailcloak plugin

* Fix comment

ciar4n committed:
Fix subform.repeatable-table multi field styling (#20209)

* Fix subform.repeatable-table multi field styling

* Fix staging less/css

* revert previous

eshiol committed:
Update FormField.php (#20168)

ReLater committed:
Update HTMLHelper.php (#19770)

klas committed:
Fix a problem with older Joomla versions language packs registration and update sites rebuild (#16355)

* Fix a problem with older language packs

Language packs installed under older Joomla versions are registered in the database under their package name, not under their name and since rebuild function fails on them.

e.g. package with 
  <name>Afrikaans (South Africa)</name>
  <packagename>af-ZA</packagename>

has name column in #_extensions table set as 	af-ZA and since existing query fails.

* Update updatesites.php

infograf768 committed:
missing single quote (#21213)

infograf768 committed:
[3.x FIX] Getting the correct associations in sidebyside view for contact and newsfeeds (#21180)

* [3.x FIX] Getting the correct associations in sidebyside view for
contact and newsfeeds

* undefined property

* correction in js

Commit:
[com_menus] Remove useglobal (#21095)

julienV committed:
[fix] fixed PluginHelper import of xtd-editors plugins (#17907)

* [fix] fixed PluginHelper import of xtd-editors plugins

* [imp] small adjustment and comment

okonomiyaki3000 committed:
Fix errors exposed by strict (#12544)

* Don't need a closure, jQuery ready function is enough. Use strict.

* Fix errors exposed by strict. Undeclared variables. wrong regex backreference

* Fix some undeclared/wrongly declared variables

* Avoid some redundant calls of $(this)

* Remove some redundant (also wrong) code.

wojsmol committed:
[3.8] Make the text clearer (#20940)

* make the text clealer

Make the PLG_TWOFACTORAUTH_TOTP_STEP3_TEXT text clealer.

* thans @brianteeman

Commit:
Adding lithuan language files.

Commit:
Updating lithuanian language strings for TinyMCE

brianteeman committed:
Appended input fields (#21257)

PR for #12448

## Steps to reproduce the issue

Effects all appended input fields. Eg. Navigate to a single article menu item (Menus -> MainMenu -> Home) and resize the screen below 767px. Input fields extend outside the viewport.

brianteeman committed:
CSS for flags (#21254)

Fix for #13678 reported by @gwsdesk

salbertson committed:
Add "Reviewed by Hound" badge (#21263)

Commit:
Adjust doc block to ensure file is included in patch packages due to accidental deletion with last release's post-update script

Commit:
Prepare 3.8.11 release candidate

Commit:
Reset to dev

Commit:
Prepare 3.8.11 release

Commit:
Bump to 3.8.12-dev

SniperSister committed:
Added security scan tool to CI setup (#20796)

* Added security scan tool to CI setup

* Updated ordering

* fixed command name

jeckodevelopment committed:
Update readme to link to Launch instead of Demo (#21364)

* No more demo

* No more demo

* Free hosting service

* Free Hosting Service

Commit:
[mod_articles_latest/news] Featured articles not shown (#21336)

Commit:
[com_content] Redundant access check (#21169)

* [com_content] Redundant access check

* Cleanup

* Cleanup

Commit:
Exclude components in discover state from list (Replaces #12469) (#21010)

* Update recaptcha.php

* Update recaptcha.php

Changing load order and adding a comment.
Proposed by @piotr-cz

* Simple CS for travis

* Update helper.php

Exclude components in discover state from list

* Update helper.php

Prevent disabled extensions "enabled = 0"

* Update helper.php

okonomiyaki3000 committed:
Fix minicolor initialization in subforms. etc. (#12994)

jeckodevelopment committed:
Create CODE_OF_CONDUCT.md (#21383)

Added our Code of Conduct as suggested by GitHub.

jeckodevelopment committed:
Removing CODE_OF_CONDUCT (#21385)

As requested in https://github.com/joomla/joomla-cms/pull/21383

brianteeman committed:
CoC Fix the english (#21386)

brianteeman committed:
Readme (#21393)

Removes link to jenkins that is no longer used

infograf768 committed:
[New Feature] Multilingual: Propagating existing associations if desired (#21321)

* [New Feature] Multilingual: Propagating existing associations if desired

* cs

* lang change

* Change and add new messages

* lang changes

* Small corrections without effect on resulting funtionnality

* using new API for token

* modifying js function to use Joomla. instead of window.

Commit:
Revert "[New Feature] Multilingual: Propagating existing associations if desired (#21321)" (#21408)

This reverts commit 8bacca95c2247a91230e56ae2d4a96a98e1ca0dc.

csthomas committed:
Fix unit tests for PostgreSQL 10 (#21370)

csthomas committed:
Escaping float value should be non-locale aware (#20643)

* Escaping float value should be non-locale aware

* Fix sqlite query dateAdd and add unit tests

okonomiyaki3000 committed:
Fix errors exposed by strict and also improve js (#20221)

* Don't need a closure, jQuery ready function is enough. Use strict.

* Radio buttons and tooltips work in subforms.

* Make chose colors work in dynamically added Chosen (in new subform rows, for example)

* Combine the two init functions as one. No reason to call two functions here.

csthomas committed:
Repair the update of database schema changes on postgreSQL (#19707)

* Repair the update of database schema changes on postgreSQL

* Add a better regex to split actions by comma

* Remove useless code

* Add a comment to sql file

rdeutz committed:
build a not on windows group (#21633)

ciar4n committed:
Stick permissions tab on scroll (#21637)

rdeutz committed:
add a note why we have a not-on-windows group

rdeutz committed:
I will use tabs, I will use tabs, I will use tabs, ….

wojsmol committed:
Update CODEOWNERS after github nick chenge (#21688)

wojsmol committed:
Add .editorconfig to this repository (#21686)

okonomiyaki3000 committed:
Set option by mime type, not by mode name (#21515)

Commit:
[com_content] Tags not shown in some cases (#21275)

* [com_content] Tags not shown in some cases

* Featured view

* CS

Commit:
Fix validation messages in reverse order on installation (#20850)

* Reverse error messages

* Move comment

wilsonge committed:
Cleanup batch copy of items (#13501)

* Cleanup batch copy of items

* Fix codestyle

* Banners doesn't use a title field

* Fix merge conflict

* Fix batch copy not copying fields

wilsonge committed:
Better documentation for addAttachment method. Fixes #14285

infograf768 committed:
[Fix] Batch menu items: Translate the menu item title when client is (#21748)

administrator

wilsonge committed:
Use the fixed menu item for open search url (#21713)

* Use the fixed menu item for open search url

* Fix Itemid

wilsonge committed:
Allow filtering by archived in Redirect Manager (#21673)

* Allow filtering by archived state in redirect component

* Fix spaces

Commit:
Don't show module chromes of disabled templates (#21437)

Commit:
[com_tags] All Tags default layout (#21031)

bembelimen committed:
Merge Script Options recursive (#19972)

frankmayer committed:
Work on indexer (#12253)

* Merge some unset()

* Lots of work on the indexer itself:

- Type safe comparisons
- a few simplifications
- a few performance optimizations

* Some Elvis's and a missing @since annotation

* One more StringHelper...

* Codestyle...

* Not to potentially break BC, reversed back to `!empty($var)` operation but with the non inverted style `(bool) $var`

* Formatting

* Fix formatting

* Revert this change for now. Look into it later again.

* Codesniffer...

* Removed repeated `use`

* Some extra work according to @andrepereiradasilva's notes

* Additional changes according to reviewer's comments

* Code style

* Fix misbehavior due to not correctly handling the single object case of mixed variable (array|object)

* Pass float value to db->escape as float may produce problems for some locales.

* "Make values float again" - (pun intended)

brianteeman committed:
Tab not slider (#21694)

* Tab not slider

As probably the majority of people use isis and not hathor templates AND we have almost told people not to use hathor any more then we should use the correct language when describing a tab. It was only a "slider" inn joomla 2.5 and hathor but its clearly a tab in isis

* grammar

wilsonge committed:
Remove debug from calendar field as it breaks saving to the DB (#21717)

Commit:
Prepare 3.8.12 release candidate

Commit:
Reset for dev

Commit:
Prepare 3.8.12 release

Commit:
Prepare 3.8.13 release

mbabker committed:
Create a class which is a decorator for a service from the DI container that is an event listener that is lazy loaded

mbabker committed:
Migrate the event listener for the metadata manager to a separate class

brianteeman committed:
initial commit

brianteeman committed:
initial commit

C-Lodder commented:
Try the following:

```
const SkipToConfig = {
	"settings": {
		"skipTo": {
			// config options for text
			"buttonLabel" : Joomla.Text._('PLG_SYSTEM_SKIPTO_SKIP_TO'),
			buttonLabel:    'Skip To...',
			menuLabel:      'Skip To and Page Outline',
			landmarksLabel: 'Skip To',
			headingsLabel:  'Page Outline',
		}
	}
};
```

Of course replacing the other static text strings with `Joomla.Text._('xxx')`

brianteeman commented:
awesome thanks @C-Lodder

brianteeman committed:
[4.0] Disable status module links (#23752)

dgrammatiko commented:
```
            Factory::getDocument()->addScriptOptions(
	                'settings', 
	                array (
	                    'buttonLabel:'      => Text::_('PLG_SYSTEM_SKIPTO_SKIP_TO'),
	                    'buttonDivTitle:'   => Text::_('PLG_SYSTEM_SKIPTO_SKIP_TO_KEYBOARD'),
	                    'menuLabel:'        => Text::_('PLG_SYSTEM_SKIPTO_SKIP_TO_AND_PAGE_OUTLINE'),
	                    'landmarksLabel:'   => Text::_('PLG_SYSTEM_SKIPTO_SKIP_TO'),
	                    'headingsLabel:'    => Text::_('PLG_SYSTEM_SKIPTO_PAGE_OUTLINE'),
	                    'contentLabel:'     => Text::_('PLG_SYSTEM_SKIPTO_CONTENT'),
	                )
	             );
````

brianteeman committed:
update

brianteeman committed:
update

brianteeman commented:
It looks from the code it should be 

```
 	"settings": {
		"skipTo": {
```

???

dgrammatiko commented:
@brianteeman since that script is outdated (es5 unmaintained) I'll propose to just port it directly yo Joomla with the appropriate attributes in the doc block. If that is something people are ok I'll do the porting

dgrammatiko commented:
@brianteeman what @C-Lodder posted is JS you need to pass the values from the PHP, use the code I provided

brianteeman commented:
@dgrammatiko they accepted a pr from @wilsonge and someone else this year ;)

brianteeman commented:
But if you were to create a dedicated joomla version then it would be awesome if it had the extra  features that the facebook skipto has.

dgrammatiko commented:
@brianteeman sorry I've missed that. Here is my approach to this:
![screenshot 2019-02-27 at 20 49 02](https://user-images.githubusercontent.com/3889375/53518407-3eeff580-3ad1-11e9-933a-fd00484cfb1a.png)

Create a PR in their repo and just add next to window.wordpress, the value `window.Joomla && typeof window.Joomla.getOptions === 'function' && window.Joomla.getOptions('skipto-settings')`
Also change the `'settings'` in the php needs to be renamed to `'skipto-settings'`

dgrammatiko commented:
> if it had the extra features that the facebook skipto has

Do you have a link for that?

brianteeman commented:
The facebook code is not available - you can see it by typing Alt /

dgrammatiko commented:
wow cool: 
![screenshot 2019-02-27 at 21 05 50](https://user-images.githubusercontent.com/3889375/53519411-824b6380-3ad3-11e9-951c-ba64990d2e54.png)

So the only difference I see id that FB also has another menu with some routes (urls) some help links. Am I reading this correctly?

zwiastunsw commented:
Am I doing something wrong? I can't install the plugin. I see it in Discovery. The Install command causes an error:
![error](https://user-images.githubusercontent.com/29527200/53525192-7f576f80-3ae1-11e9-9836-d11862a001e2.png)

brianteeman commented:
It is still a work in progress

SharkyKZ commented on plugins/system/skipto/skipto.xml:
```diff
@@ -0,0 +1,19 @@
+	<version>4.0.0</version>
+	<description>PLG_SYSTEM_SKIPTO_XML_DESCRIPTION</description>
+	<files>
+		<filename plugin="system">skipto.php</filename>
```
Apply this now please so the plugin can be discovered.
```suggestion
		<filename plugin="skipto">skipto.php</filename>
```

brianteeman replied:
Thanks - I did notice that last night. This is only a draft pr and I actually only shared it so that I could get some js help

brianteeman committed:
xml

brianteeman committed:
xml

brianteeman commented:
@dgrammatiko yes thats about it

dgrammatiko committed:
[4.0] Field media custom element clean up (#23882)

brianteeman committed:
comments

brianteeman committed:
comments

brianteeman committed:
cs

brianteeman committed:
cs

zwiastunsw commented:
If it is not a supplement from the "system" group, but from the "skipto" group, maybe the language strings should be changed as well? (PLG_<del>SYSTEM_</del>SKIPTO_SKIP_TO , etc.),

brianteeman commented:
@zwiastunsw at the moment I cant get the script or css to load at all :(

@dgrammatiko I must be doing something obviously wrong but I cant see why this isn't doing anything
```
			HTMLHelper::_('script', 'vendor/skipto/js/skipTo.js', ['version' => 'auto', 'relative' => true]);
			HTMLHelper::_('stylesheet', 'vendor/skipto/css/SkipTo.css', ['version' => 'auto', 'relative' => true], ['defer' => true]);
```

brianteeman commented:
@zwiastunsw the language string constants are correct. Look at the other system plugin language files and you will see they follow the PLG_SYSTEM_NAME_ convention

zwiastunsw commented:
I've already noticed - I understood the change in the xml file, sory

dgrammatiko commented:
@brianteeman you need to init the script, try:
```
Factory::getDocument()->addScriptDeclaration(
"
window.skipToMenuInit(Joomla.getOptions('skipto-settings'));
"
);
```
Also you need to change the `settings` part in https://github.com/joomla/joomla-cms/pull/24042/files#diff-da83472045fbc26e0c9e98f68224eeedR56
to 'skipto-settings'. That should do it

wilsonge committed:
[4.0] Update composer deps (#24037)

brianteeman commented:
@dgrammatiko that may work - I dont know. the problem is that the two lines that should load the css and js are just not doing anything. no 404, nothing, nada, zilch

Must be something obvious but I cant see it

dgrammatiko commented:
@brianteeman try this:
```
			// Add strings for translations in Javascript.
			Factory::getDocument()->addScriptOptions(
				'skipto-settings',
					[
						'settings' => [
							'skipTo' => [
								'buttonLabel:'      => Text::_('PLG_SYSTEM_SKIPTO_SKIP_TO'),
								'buttonDivTitle:'   => Text::_('PLG_SYSTEM_SKIPTO_SKIP_TO_KEYBOARD'),
								'menuLabel:'        => Text::_('PLG_SYSTEM_SKIPTO_SKIP_TO_AND_PAGE_OUTLINE'),
								'landmarksLabel:'   => Text::_('PLG_SYSTEM_SKIPTO_SKIP_TO'),
								'headingsLabel:'    => Text::_('PLG_SYSTEM_SKIPTO_PAGE_OUTLINE'),
								'contentLabel:'     => Text::_('PLG_SYSTEM_SKIPTO_CONTENT'),
							]
						]
					]
			);
			HTMLHelper::_('script', 'vendor/skipto/skipTo.js', ['version' => 'auto', 'relative' => true]);
			HTMLHelper::_('stylesheet', 'vendor/skipto/SkipTo.css', ['version' => 'auto', 'relative' => true]);

			$document->addScriptDeclaration("window.skipToMenuInit(Joomla.getOptions('skipto-settings'));");
```

Hackwar committed:
[4.0] Smart Search: Adding nested taxonomy (#21327)

C-Lodder commented:
@brianteeman silly question, but did you run `npm i`?

brianteeman commented:
@C-Lodder yes  and it created
media/vendor/skipto/js/skipTo.js
media/vendor/skipto/css/SkipTo.css

if the files were not there then I would have expected to get a 404 or something. but the css and js are not even in the source of the generated page

dgrammatiko commented:
@brianteeman @C-Lodder the error is on the relative path and the fact that the path given was absolute. The snippet above should fix that

brianteeman commented:
yes - looks like progress. - thanks

I will carry on working on this now

C-Lodder commented:
@dgrammatiko ah yes, didn't even notice the `js/css` in the path

infograf768 committed:
[4.0][com_media] RTL: adapting media manager to rtl + some (#24043)

bembelimen committed:
[4.0] Update bootstrap version (#24003)

brianteeman committed:
add dropmenu

brianteeman committed:
add dropmenu

C-Lodder commented:
@brianteeman - Just seen you've added a dropdown menu script. Would you be able to try using the Custom Element, as I've done here: https://github.com/joomla/joomla-cms/pull/23665/files ?

Only thing missing from my PR is fetching it via NPM

brianteeman commented:
@C-Lodder I added it because I realised the paypal script came with it. They combine it into the skipto.js file in their build tools but we arent using the tools so I assumed I needed to do it this way.

More than happy to use anything you want me to use. 

But right now I am still getting js errors
```
Uncaught TypeError: Cannot read property 'nodeType' of null
Uncaught TypeError: window.skipToMenuInit is not a function
```

As you can tell I really dont have a clue with js

C-Lodder commented:
I'm more than happy to help when I get back from work in a couple of hours

brianteeman commented:
Thanks - I am out this evening but you can always do a pr on my branch

alikon committed:
Fix default date for postgresql (#24046)

C-Lodder commented:
In the mean time, it looks like you're missing some config options:

https://github.com/paypal/skipto#configuring-skipto-options

such as the ID's. So that would probably explain the first error above.

brianteeman commented:
From what I can tell the defaults are being used which should be ok - anyway moving on to work on something else for now

brianteeman committed:
[4.0] Toolbar button consistency (#24048)

All the toolbars are ordered "new - status - batch - rebuild"

The toolbar for menus was "new - status - rebuild - batch"

This simple PR makes them all consistent

alikon committed:
com_finder update (#24047)

chmst committed:
[4.0] Group toolbar buttons for workflow stages and transitions (#23608)

Co-Authored-By: chmst <dev@chmst.de>

amitranjan2 committed:
[4.0] An error has occurred on Search Filter (#23679)

brianteeman committed:
Global checkin - select call style (#24031)

The class text-center was missing from the select all box

wilsonge committed:
Merge commit '20eeef5' into 4.0-dev

wilsonge committed:
Merge commit '9007811' into 4.0-dev

wilsonge committed:
Merge commit '7b73d5f' into 4.0-dev

wilsonge committed:
Merge commit '044e044' into 4.0-dev

wilsonge committed:
Merge commit '603db7e' into 4.0-dev

wilsonge committed:
Fix stability

wilsonge committed:
Merge commit '1f6d87a' into 4.0-dev

wilsonge committed:
Merge commit '1dc9d91' into 4.0-dev

wilsonge committed:
Merge pull request #24068 from wilsonge/3811-merge

Merge 3.8.11 Release into Joomla 4

wilsonge committed:
Merge commit '8116f00' into 4.0-dev

wilsonge committed:
Merge branch '4.0-dev' of github.com:joomla/joomla-cms into 4.0-dev

wilsonge committed:
Merge commit 'f3094f2' into 4.0-dev

wilsonge committed:
Merge pull request #24069 from wilsonge/3812-merge

Start of 3.8.12 merge

wilsonge committed:
Merge commit '2b5d12a' into 4.0-dev

wilsonge committed:
Merge commit '27ca70a' into 4.0-dev

wilsonge committed:
Merge branch '4.0-dev' of github.com:joomla/joomla-cms into 4.0-dev

wilsonge committed:
Fix codestyle in mod search, move calendar locale to new location

wilsonge committed:
Merge pull request #24072 from wilsonge/3812-merge-2

Merge in the rest of Joomla 3.8.12 and all 1 commit of 3.8.13

wilsonge committed:
Fix path to file (#24074)

wilsonge committed:
[4.0] Fix webcomponents not being included on the page (#24075)

brianteeman committed:
[4.0] Remove com_admin profile (#24064)

The users icon in the mod_status module links to a view of the logged in users profile. However because it is NOT using com_users any plugins that extend com_users are not displayed. For example TFA plugins when enabled are not accessible from this link and are only accessible from the users component. Maintaining this additional view makes no sense to me. Its a regular cause of bug reports etc.

This PR completely removes the com_admin profile and replaces the link in the module to use com_users. Now any plugins that extend com_users are accessible from the link which is what the user expects

amitranjan2 committed:
Fix Finder logging table creation in MySQL (#24060)

brianteeman committed:
[4.0] Change Status dropdown consistency (#24078)

Having a consistent UI is essential. All off the new change status dropdown items are ordrered Published->Unpublished->Archive->Checkin->Trashed except for com_content where Archive and Checkin are reversed.

This PR changes the order of the articles change status dropdown items to be consistent with all the others

puneeth-chanda committed:
[4.0]Table text changed (#24022)

wilsonge committed:
Fix typehint to be compatible with parent interface (#24041)

brianteeman committed:
add defer to the scripts so that they actually load

brianteeman committed:
add defer to the scripts so that they actually load

brianteeman committed:
remove extra :

brianteeman committed:
remove extra :

brianteeman commented:
Fixed one of the errors 
`Uncaught TypeError: Cannot read property 'nodeType' of null
`
by adding defer 

The skipto toolbar now loads and works
But the second problem still exists which prevents customisation
`Uncaught TypeError: window.skipToMenuInit is not a function`

dgrammatiko commented:
@brianteeman if you type `skipToMenuInit()` in the browser's console what are you getting?

brianteeman commented:
undefined

dgrammatiko committed:
[4.0] Fix debug (#23637)

SharkyKZ committed:
[4.0] Fatal error when editing Filesystem - Local plugin (#24103)

Thanks

bahl24 committed:
[4.0]Added lang string for upload file (#23951)

* lang string

Signed-off-by: Nitish Bahl <nitishbahl24@gmail.com>

* added in default_texts.php

Signed-off-by: Nitish Bahl <nitishbahl24@gmail.com>

* Alpha ordering of strings

* Update default_texts.php

* Adding string in frontend

* oops

bahl24 committed:
[4.0]com_media missing download lang string (#23967)

* initialized download string

Signed-off-by: Nitish Bahl <nitishbahl24@gmail.com>

* changed to download item

wilsonge committed:
Add Webservices to Joomla (#23424)

C-Lodder committed:
[4.0] Toolbar custom element improvements (#23767)

wilsonge committed:
Fix missing string for webservices login permission (#24121)

dgrammatiko committed:
Cleanup user form field (#23881)

zwiastunsw commented:
I tested it. It works well. 
My questions. Will the add-on be configurable? 
Will it be able to be enabled on the front page? 
Also, are you considering adding accessibility help links (just like on Facebook)?

brianteeman commented:
>Will the add-on be configurable?

Once someone helps me work out why the current options isnt working

> Will it be able to be enabled on the front page?

It could be. I was thinking - admin - on by default, site on by option

> Also, are you considering adding accessibility help links (just like on Facebook)?

would love to but that would require creating our own version of the script

zwiastunsw commented:
My suggestion: merge the plugins as it is now, to work on its usefulness (adding landmarks, adding headers on backend pages, etc.).

brianteeman commented:
I really don't want to merge it until at least the translation is working

wilsonge committed:
Small webservice related fixes (#24133)

SharkyKZ committed:
[4.0] Remove references to non-existing classes (#24124)

wilsonge committed:
Fix reference to bad constant in metadata manager (#24120)

wilsonge committed:
Remove path that no longer exists

This directory was removed in https://github.com/joomla/joomla-cms/pull/23739

brianteeman committed:
[4.0] Namespace JMailHelper (#24119)

Fix the last namespace update for JMailHelper

wilsonge commented on plugins/system/skipto/skipto.php:
```diff
@@ -0,0 +1,77 @@
+		HTMLHelper::_('script', 'vendor/skipto/skipTo.js', ['version' => 'auto', 'relative' => true], ['defer' => true]);
+		HTMLHelper::_('stylesheet', 'vendor/skipto/SkipTo.css', ['version' => 'auto', 'relative' => true]);
+
+		$document->addScriptDeclaration("window.skipToMenuInit(Joomla.getOptions('skipto-settings'))");
```
```
$document->addScriptDeclaration("document.addEventListener('DOMContentLoaded', function() {
    window.SkipToConfig = Joomla.getOptions('skipto-settings');
    window.skipToMenuInit();
});");
```

wait for the dom to load before you call it and you should be good to go

brianteeman replied:
testing

brianteeman committed:
js - thanks @wilsonge

brianteeman committed:
js - thanks @wilsonge

brianteeman commented:
thanks to @wilsonge we dont get any javascript errors now.

still dont get the settings being used though :(

I will take another look tomorrow

wilsonge commented:
Sorry I edited my code to fix that. Then the settings are used

wilsonge commented on plugins/system/skipto/skipto.php:
```diff
@@ -0,0 +1,79 @@
+
+		if ($this->app->isClient('administrator'))
+		{
+		// Add strings for translations in Javascript.
```
So for reasons I totally don't understand to make language files load i had to addtionally call `$this->loadLanguage()` here to actually force the language files to load (despite that's what `autoloadLanguage` should do above). But it's been a long day travelling/working so I'm going to pretend I'm doing something stupid :)

wilsonge commented:
Btw. How do you actually activate this thing. I can see it in the dom. just can't see any way to visually get it on screen. Or is that the idea?

brianteeman commented:
Hit the tab key or alt 0

On Sat, 9 Mar 2019, 00:19 George Wilson, <notifications@github.com> wrote:

> Btw. How do you actually activate this thing. I can see it in the dom.
> just can't see any way to visually get it on screen. Or is that the idea?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/joomla/joomla-cms/pull/24042#issuecomment-471122773>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABPH8SMa1ZdRUfdGo1cmWu8VipQEZ4kAks5vUv34gaJpZM4bVRRy>
> .
>

dgrammatiko commented:
@brianteeman try
```js
$document->addScriptDeclaration("document.addEventListener('DOMContentLoaded', function() {
    window.skipToMenuInit(Joomla.getOptions('skipto-settings'));
});");
```

wilsonge committed:
Tag Joomla 4 Alpha 7

brianteeman committed:
[4.0] Postgres default date (#24136)

* [4.0] Postgresql date fix

Correct the date used in a postgres sql file

* installation

wilsonge commented:
> Hit the tab key or alt 0

OK. Either way - make the javascript  change I suggested - @dgrammatiko 's is wrong (that function doesn't take an options argument)

```
$document->addScriptDeclaration("document.addEventListener('DOMContentLoaded', function() {
    window.SkipToConfig = Joomla.getOptions('skipto-settings');
    window.skipToMenuInit();
});");
```

Then see if the strings translate for you or not

wilsonge committed:
Bump version back to dev

brianteeman commented:
I will do after the rugby #priorities

wilsonge commented:
It's only important if scotland manage to upset :D :D

brianteeman committed:
js

brianteeman committed:
js

brianteeman commented:
@wilsonge Thanks!!!!

The javascript is now working and like you I have had to add the $this->loadLanguage() 

I am going to work on adding options etc in the plugin now

brianteeman committed:
add config setting for site/admin/both - codestyle from yubikey plugin

brianteeman committed:
add config setting for site/admin/both - codestyle from yubikey plugin

Quy commented on plugins/system/skipto/skipto.php:
```diff
@@ -0,0 +1,106 @@
+			$current_section = 0;
+		}
+
+		if (!($current_section & $section))
```
```suggestion
		if (!($current_section && $section))
```

brianteeman committed:
Merge remote-tracking branch 'brian/npmskipto' into npmskipto

brianteeman commented:
See #24142

dgrammatiko commented:
@wilsonge yeah for sure I'm wrong:
<img width="922" alt="Screenshot 2019-03-09 at 22 21 03" src="https://user-images.githubusercontent.com/3889375/54077498-b2f08180-42b9-11e9-9b01-6f32176e25ae.png">
